WD-Firmware/rtos/rtosal/src/rtosal_int_vect.S

```diff
@@ -73,77 +73,24 @@
 /*
 This implementation supports riscv privileged v1.10
 */
-.include "psp_asm_macros.h"
-
-/* saves mstatus and tcb */
-.macro m_portSAVE_CONTEXT
-    .global    pxCurrentTCB
-    /* Store mstatus */
-    csrr      t0, mstatus
-    m_STORE   t0, 32 * REGBYTES(sp)
-    /* Store current stackpointer in task control block (TCB) */
-    m_LOAD    t0, pxCurrentTCB
-    m_STORE   sp, 0x0(t0)
-.endm
-
- /* restore mstatus and tcb */
-.macro m_portRESTORE_CONTEXT
-    .global    pxCurrentTCB
-    /* Load stack pointer from the current TCB */
-    m_LOAD    sp, pxCurrentTCB
-    m_LOAD    sp, 0x0(sp)
-    /* Load task program counter */
-    m_LOAD    t0, 33 * REGBYTES(sp)
-    csrw      mepc, t0
-    /* Load saved mstatus */
-    m_LOAD    t0, 32 * REGBYTES(sp)
-    csrw      mstatus, t0
-.endm
-
-.macro m_portEND_SWITCHING_ISR branch_label
-    /* save address of g_rtosalContextSwitch -> a0 */
-    la        a0, g_rtosalContextSwitch
-    /* load the value g_rtosalContextSwitch -> a1 */
-    m_LOAD    a1, 0x0(a0)
-    /* check if g_rtosalContextSwitch is set - need to do context switch */
-    beqz      a1, \branch_label
-    /* clear g_rtosalContextSwitch */
-    /* TODO: if bitmanip exist add bit set */
-    m_STORE   zero, 0x0(a0)
-    /* perform context switch */
-.if D_USE_FREERTOS
-    m_portSAVE_CONTEXT
-    jal     vTaskSwitchContext
-    m_portRESTORE_CONTEXT
-.elseif D_USE_THREADX
-.endif /* .if D_USE_FREERTOS */
-.endm
-
-/* Saves current Machine Exception Program Counter (MEPC) as task program counter */
-.macro m_portSAVE_EPC
-    csrr      t0, mepc
-    m_STORE   t0, 33 * REGBYTES(sp)
-.endm
-
-/* Saves current return adress (RA) as task program counter */
-.macro m_portSAVE_RA
-    LOAD      t0, 1 * REGBYTES(sp)
-    m_STORE   t0, 33 * REGBYTES(sp)
-.endm
-
-/*/not called on a context switch*/
-.macro m_portRESTORE_SP
-    m_LOAD    x2, 2 * REGBYTES(sp)
-.endm
+
+#include "psp_asm_macros.inc"
```

Comment:
remove #include -> to be .include

Reply:
Done. Currently contains only rtosalInstallIsr function that actually does nothing